packages-exp/auth-exp/src/core/strategies/custom_token.test.ts

```diff
@@ -67,31 +71,59 @@ describe('core/strategies/signInWithCustomToken', () => {
   });
   afterEach(mockFetch.tearDown);
 
-  it('should return a valid user credential', async () => {
-    const {
-      user,
-      operationType,
-      _tokenResponse
-    } = (await signInWithCustomToken(
-      auth,
-      'look-at-me-im-a-jwt'
-    )) as UserCredentialInternal;
-    expect(_tokenResponse).to.eql(idTokenResponse);
-    expect(user.uid).to.eq('local-id');
-    expect(user.displayName).to.eq('display-name');
-    expect(operationType).to.eq(OperationType.SIGN_IN);
+  describe('#signInWithCustomToken', () => {
+    it('should return a valid user credential', async () => {
+      const { user, operationType, _tokenResponse } =
+        (await signInWithCustomToken(
+          auth,
+          'look-at-me-im-a-jwt'
+        )) as UserCredentialInternal;
+      expect(_tokenResponse).to.eql(idTokenResponse);
+      expect(user.uid).to.eq('local-id');
+      expect(user.displayName).to.eq('display-name');
+      expect(operationType).to.eq(OperationType.SIGN_IN);
+    });
+
+    it('should send with a valid request', async () => {
+      await signInWithCustomToken(auth, 'j.w.t');
+      expect(signInRoute.calls[0].request).to.eql({
+        token: 'j.w.t',
+        returnSecureToken: true
+      });
+    });
+
+    it('should update the current user', async () => {
+      const { user } = await signInWithCustomToken(auth, 'oh.no');
+      expect(auth.currentUser).to.eq(user);
+    });
   });
 
-  it('should send with a valid request', async () => {
-    await signInWithCustomToken(auth, 'j.w.t');
-    expect(signInRoute.calls[0].request).to.eql({
-      token: 'j.w.t',
-      returnSecureToken: true
+  describe('#setCustomTokenProvider', () => {
+    it('sets a custom token provider', () => {
+      const provider = {
+        async getCustomToken() {
+          return Promise.resolve('');
+        }
+      };
+
+      setCustomTokenProvider(auth, provider);
+
+      expect(auth._customTokenProvider).to.eql(provider);
     });
   });
 
-  it('should update the current user', async () => {
-    const { user } = await signInWithCustomToken(auth, 'oh.no');
-    expect(auth.currentUser).to.eq(user);
+  describe('#clearCustomTokenProvider', () => {
+    it('clears the custom token provider', () => {
+      const provider = {
+        async getCustomToken() {
+          return Promise.resolve('');
```

Comment:
No need to `Promise.resolve()` here... the `async` keyword on the function makes that implied. You can just write:
```ts
async getCustomToken(): Promise<string> {
  return '';
}
```

Also, the getCustomToken() function should have a return type defined. I'm surprised the linter isn't complaining